073-improve-handling-of-CA-renewals-and-replacements-in-client-based-operands.md

```diff
@@ -0,0 +1,211 @@
+
+For simple CA renewals, typically both new and old CA public keys are valid at the same time (for a limited period of time - e.g. 30 days) and only one of them needs to be trusted.
+In ideal case, during the renewal, the Kubernetes Secret is updated and the new public key is put in place of the old one (under the same key, e.g. `ca.crt`).
+Strimzi will then automatically roll the operand to load the new public key.
+This is how it works for example when the operand is connected to a Strimzi-based Apache Kafka cluster running in the same namespace.
+You can use the public key Secret that is part of the Kafka cluster:
+```yaml
+  tls:
+    trustedCertificates:
+      - secretName: my-cluster-cluster-ca-cert
+        certificate: ca.crt
+```
+
+And everything will happen in completely automatically as described above.
+
+But if the Apache Kafka cluster is running in a different namespace, different Kubernetes cluster or is not powered by Strimzi, it might be more complicated:
+* Different Kubernetes Secrets might be used for the old and new public keys
+* The same Kubernetes Secret might be used, but the new key might be stored under a different key inside the Secret
+
+That means the user has to edit the custom resource at the right time and add the new certificate to the list of the trusted certificates.
+
+Things get even more complicated when the CA is completely replaced.
+In such case, the operand needs to typically trust both the old and new CA at the same time because while the Apache Kafka cluster is migrating from the old CA to the new CA different brokers might use server certificates from both CAs at the same time.
+If you migrate to the new CA manually, you might still be able to orchestrate everything by editing the custom resource and modifying the trusted certificates as needed.
+
+But when the whole process is automated, it might not be easy to orchestrate it manually.
+This applies even to the ideal situation when the Apache Kafka cluster is based on Strimzi and runs in the same namespace and user for example triggers the CA replacement using the `strimzi.io/force-replace` annotation.
+What happens is the following scenario:
+1. Strimzi generates the new CA public and private keys and stores the new CA public key as `ca.crt` in the Kubernetes Secret.
+   The public key of the old CA is renamed to `ca-<timestamp>.crt`.
+2. Strimzi starts rolling the Kafka brokers to make sure they trust the new CA.
+   At this point in time, the Kafka brokers still use the old server certificates signed by the old CA.
+3. The Kafka client based operand (e.g. Kafka Connect cluster) using the typical `.spec.tls` configuration:
+   ```yaml
+   tls:
+      trustedCertificates:
+         - secretName: my-cluster-cluster-ca-cert
+           certificate: ca.crt
+   ```
+   Will see that the `ca.crt` in the Secret was updated and will trigger a rolling update of the operand to use the new CA.
```

Comment:
No, because in the current code, it cares only about the specified field `ca.crt`. It completely ignores any newly added fields such as the `ca-<timestamp>.crt`.